src/authenticationservices.cs

```diff
@@ -192,5 +227,481 @@ interface ASWebAuthenticationSession {
+		void BeginAuthorization (ASAuthorizationProviderExtensionAuthorizationRequest request);
+
+		[Export ("cancelAuthorizationWithRequest:")]
+		void CancelAuthorization (ASAuthorizationProviderExtensionAuthorizationRequest request);
+	}
+
+	// [Unavailable (PlatformName.UIKitForMac)]
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	interface ASAuthorizationProviderExtensionAuthorizationRequest {
+
+		[Export ("doNotHandle")]
+		void DoNotHandle ();
+
+		[Export ("cancel")]
+		void Cancel ();
+
+		[Export ("complete")]
+		void Complete ();
+
+		[Export ("completeWithHTTPAuthorizationHeaders:")]
+		void Complete (NSDictionary<NSString, NSString> httpAuthorizationHeaders);
+
+		[Export ("completeWithHTTPResponse:httpBody:")]
+		void Complete (NSHttpUrlResponse httpResponse, NSData httpBody);
+
+		[Export ("completeWithError:")]
+		void Complete (NSError error);
+
+		[Async]
+		[Export ("presentAuthorizationViewControllerWithCompletion:")]
+		void PresentAuthorizationViewController (Action<bool, NSError> completion);
+
+		[Export ("url")]
+		NSUrl Url { get; }
+
+		[Export ("requestedOperation")]
+		string RequestedOperation { get; }
+
+		[Export ("httpHeaders")]
+		NSDictionary<NSString, NSString> HttpHeaders { get; }
+
+		[Export ("httpBody")]
+		NSData HttpBody { get; }
+
+		[Export ("realm")]
+		string Realm { get; }
+
+		[Export ("extensionData")]
+		NSDictionary ExtensionData { get; }
+
+		[Export ("callerBundleIdentifier")]
+		string CallerBundleIdentifier { get; }
+
+		[Export ("authorizationOptions")]
+		NSDictionary AuthorizationOptions { get; }
+	}
+
+	[Watch (6,0), TV (13,0), Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationRequest : NSCopying, NSSecureCoding {
+
+		[Export ("provider", ArgumentSemantic.Strong)]
+		IASAuthorizationProvider Provider { get; }
+	}
+
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationSingleSignOnCredential : ASAuthorizationCredential {
+
+		[NullAllowed, Export ("state")]
+		string State { get; }
+
+		[NullAllowed, Export ("accessToken", ArgumentSemantic.Copy)]
+		NSData AccessToken { get; }
+
+		[NullAllowed, Export ("identityToken", ArgumentSemantic.Copy)]
+		NSData IdentityToken { get; }
+
+		[Export ("authorizedScopes", ArgumentSemantic.Copy)]
+		[BindAs (typeof (ASAuthorizationScope []))]
+		NSString[] AuthorizedScopes { get; }
+
+		[NullAllowed, Export ("authenticatedResponse", ArgumentSemantic.Copy)]
+		NSHttpUrlResponse AuthenticatedResponse { get; }
+	}
+
+	[NoWatch, NoTV, Mac (10,15, onlyOn64: true), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationSingleSignOnProvider : ASAuthorizationProvider {
+
+		[Static]
+		[Export ("authorizationProviderWithIdentityProviderURL:")]
+		ASAuthorizationSingleSignOnProvider CreateProvider (NSUrl url);
```

Comment:
The parameter could be `identityProviderUrl` to give more context to people using this API.